TheengsGateway/ble_gateway.py

```diff
@@ -459,6 +459,14 @@ def decode_advertisement(
 
         if decoded_json:
             decoded_json = json.loads(decoded_json)
+
+            if "prmac" in decoded_json:
+                address = decoded_json.get("id")
+                if address in self.configuration["identities"]:
+                    decoded_json.pop("prmac", None)
+                else:
+                    decoded_json["type"] = "RMAC"
```

Comment:
You can turn this into a separate method of the `Gateway` class to lift the logic out of the already long `decode_advertisement` method.